File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        map = {}
        k = 0

        for i in range(len(nums)):
            n = nums[i]
            map[n] = map[n] + 1 if n in map else 1

            if (i == 0) or (n in map and map[n] < 3):
                nums[k] = n
                k += 1
        
        return k

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 0;
        int value;
        Map<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            value = map.containsKey(nums[i]) ? map.get(nums[i]) + 1 : 1;
            map.put(nums[i], value);
            
            if(i == 0 || (map.containsKey(nums[i]) && map.get(nums[i]) < 3)){
                nums[k] = nums[i];
                k += 1;
            }
        }
        return k;
    }
}